### PR TITLE
refactor: allow implicit allowed roles/locales

### PR DESCRIPTION
**Pull request by plmercereau**
Before submitting this PR:

### Checklist

- [x] No breaking changes
- [x] Tests pass
- [x] New features have new tests
- [x] Documentation is updated

### CODE REVIEW

**Comment by dbarrosop** [comment]
+1 from a functionality point of view

File: .changeset/quiet-berries-develop.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+Now, the following configuration will also work:
+
+```
+AUTH_USER_DEFAULT_ROLE=user
+AUTH_USER_DEFAULT_ALLOWED_ROLES=other
+```
+
+Both syntaxes will allow the roles `['user', 'me', 'other']`
```
Comment (elitan) [inline .changeset/quiet-berries-develop.md:23]:
```suggestion
Both syntaxes will allow the roles `user`, `me`, and `other`.
```

File: .changeset/quiet-berries-develop.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+---
+'hasura-auth': patch
+---
+
+Always add required allowed roles and locales
```
Comment (elitan) [inline .changeset/quiet-berries-develop.md:5]:
```suggestion
- The default role is automatically added to the allowed roles.
- The default locale is automatically added to the allowed locales.
```

File: .changeset/quiet-berries-develop.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Until now, it was required to pass on explicilty the `"me"` and `AUTH_USER_DEFAULT_ROLE` to `AUTH_USER_DEFAULT_ALLOWED_ROLES`. They are now automatically added if they are not in the environment variable.
```
Comment (elitan) [inline .changeset/quiet-berries-develop.md:7]:
```suggestion
Previously, it was explicitly required to add the `me` and `AUTH_USER_DEFAULT_ROLE` roles to `AUTH_USER_DEFAULT_ALLOWED_ROLES`. They are now automatically added to `AUTH_USER_DEFAULT_ALLOWED_ROLES`.
```

File: .changeset/quiet-berries-develop.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+Now, the following configuration will also work:
+
+```
+AUTH_USER_DEFAULT_ROLE=user
+AUTH_USER_DEFAULT_ALLOWED_ROLES=other
+```
+
+Both syntaxes will allow the roles `['user', 'me', 'other']`
+
+Similarly, it is not required to add the ``to the`AUTH_LOCALE_ALLOWED_LOCALES`.
```
Comment (elitan) [inline .changeset/quiet-berries-develop.md:25]:
```suggestion
Similarly, it is no longer a requirement to add the value of `AUTH_LOCALE_DEFAULT` to the `AUTH_LOCALE_ALLOWED_LOCALES`.
```

File: .changeset/quiet-berries-develop.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+Now, the following configuration will also work:
+
+```
+AUTH_LOCALE_DEFAULT=en
+AUTH_LOCALE_ALLOWED_LOCALES=en,fr
+```
+
+Both syntwaxes will allow the locales `['en', 'fr']`
```
Comment (elitan) [inline .changeset/quiet-berries-develop.md:41]:
```suggestion
Both syntaxes will allow the locales `en` and `fr`.
```

File: src/utils/env.ts [hunk 6]
```diff
@@ -142,19 +142,24 @@ export const ENV = {
     return castStringEnv('AUTH_USER_DEFAULT_ROLE', 'user');
   },
   get AUTH_USER_DEFAULT_ALLOWED_ROLES() {
-    return castStringArrayEnv('AUTH_USER_DEFAULT_ALLOWED_ROLES', [
+    const allowedRoles = [
       'me',
```
Comment (elitan) [inline src/utils/env.ts:146]: In a future major release, let's consider if we should remove this hardcoded `me` role.

**Review by elitan — CHANGES_REQUESTED** [review]

**Review by elitan — APPROVED** [review]

**Pull request merged**